adam/experiment/experiment_utils.py

```diff
@@ -50,6 +55,71 @@
 )
 
 
+# Experiment Utils
+def observer_states_by_most_recent(
+    pickled_path: Path, pickel_name: str
+) -> Iterable[Tuple[int, Path]]:
+    paths = []
+    for logged_state_path in pickled_path.glob(f"{pickel_name}*.pkl"):
+        if not logged_state_path.is_file():
+            logging.warning("Skipping non-file learner state %s.", str(logged_state_path))
+            continue
+        iteration_number_string = logged_state_path.name.replace(
+            f"{pickel_name}", ""
+        ).replace(".pkl", "")
+        try:
+            iteration_number = int(iteration_number_string)
+        except ValueError:
+            logging.warning(
+                "Skipping Observer state file with bad iteration number %s.",
+                iteration_number_string,
+            )
+            continue
+        paths.append((iteration_number, logged_state_path))
+        logged_state_path.stat()
+    return sorted(paths, reverse=True)
+
+
+def restore_report_state(path: Path, num_instances: int) -> None:
+    """
+    Restore the output report files to the appropriate number of instances to match the loaded observer
+    """
+    lines_to_write_back: List[str] = []
+    with open(str(path), "r") as file:
+        for num, line in enumerate(file):
+            if num < num_instances:
+                lines_to_write_back.append(line.strip())
+
+    # Remove the old file
+    os.remove(str(path))
```

Comment:
Does it make sense to use `os.remove` over `path.unlink()` here? If there's a reason, then this should get a comment, otherwise, maybe use the method:
```suggestion
    path.unlink()
```